### PR TITLE
Fix sodium scroll crashes on Android 5.x.x

### PR DESCRIPTION
**Pull request by daividssilverio**
## What's this?
A fix for the most common crash in 1.5.1.
Also it makes the tooltips appear in on Android 5 as well.

### Relationships
Part of #3343 

Closes nothing.

## Why do we want this?
Because we care about our users with Android 5 as well.

## How is it done?
It just adds an implementation to RecyclerView.OnScrollListener since I wasn't able to fix this using the linker.

### Why not another way?
Including `mono.android.view.View_OnScrollChangeListenerImplementor` on `LinkerPleaseInclude` didn't work for me.

### Side effects
🍭 users also couldn't see the swipe onboarding tooltips when I included the scroll listener.
But I've added a simple fix for that as well.

## Review hints
Test on Android 5 on develop to manifest the crash.
Test again from this branch and everything should be fine.

## :squid: Permissions
🦑 whenever

### CODE REVIEW

File: Toggl.Giskard/ViewHelpers/SimpleRecyclerViewScrollChangeListener.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+using System.Reactive;
+using System.Reactive.Subjects;
+using Android.Support.V7.Widget;
+
+namespace Toggl.Giskard.ViewHelpers
+{
+    public sealed class SimpleRecyclerViewScrollChangeListener : RecyclerView.OnScrollListener
+    {
+        public ISubject<Unit> onScroll { get; set; }
```
Comment (willsb) [inline Toggl.Giskard/ViewHelpers/SimpleRecyclerViewScrollChangeListener.cs]: Properties are `PascalCase`

File: Toggl.Giskard/Activities/MainActivity.Onboardings.cs [hunk 2]
```diff
@@ -43,6 +43,23 @@ public partial class MainActivity
...
+        private ISubject<Unit> mainRecyclerViewScrollChanges = new Subject<Unit>();
+
+        protected override void OnResume()
+        {
+            base.OnResume();
+            var scrollListener = new SimpleRecyclerViewScrollChangeListener()
+            {
+                onScroll = mainRecyclerViewScrollChanges
+            };
+            mainRecyclerView.AddOnScrollListener(scrollListener);
```
Comment (willsb) [inline Toggl.Giskard/Activities/MainActivity.Onboardings.cs]: Why not `mainRecyclerView.Rx().OnScrolled()`, that adds a reactive listener and removes it once everything is disposed?

File: Toggl.Giskard/Activities/MainActivity.Onboardings.cs [hunk 3]
```diff
@@ -43,6 +43,24 @@ public partial class MainActivity
         private IDisposable editTimeEntryOnboardingStepDisposable;
 
         private IObservable<Unit> mainRecyclerViewChangesObservable;
+        private ISubject<Unit> mainRecyclerViewScrollChanges = new Subject<Unit>();
```
Comment (willsb) [inline Toggl.Giskard/Activities/MainActivity.Onboardings.cs]: Is this really needed? Do we really have to clean the disposable on ever pause?

File: Toggl.Giskard/ViewHelpers/SimpleRecyclerViewScrollChangeListener.cs [hunk 4]
```diff
@@ -0,0 +1,23 @@
+using System.Reactive;
+using System.Reactive.Subjects;
+using Android.Support.V7.Widget;
+
+namespace Toggl.Giskard.ViewHelpers
+{
+    public sealed class SimpleRecyclerViewScrollChangeListener : RecyclerView.OnScrollListener
+    {
+        public ISubject<Unit> OnScroll { get; set; }
```
Comment (willsb) [inline Toggl.Giskard/ViewHelpers/SimpleRecyclerViewScrollChangeListener.cs]: Why not make this immutable?

Comment (daividssilverio) [reply Toggl.Giskard/ViewHelpers/SimpleRecyclerViewScrollChangeListener.cs]: No reason whatsoever.

**Pull request merged**